### PR TITLE
Update DG with grade and comment

### PR DESCRIPTION
**Pull request by Originalidk**
Update DG with grade and comment features

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -382,7 +429,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case resumes at step 2.
+
+**Use case: Delete the grade of an assignment**
+
+**MSS**
+
+1.  User requests to list persons
+2.  F.A.K.E.J.A.R.V.I.S. shows a list of persons
+3.  User requests to delete the grade of an assignment for a specific person in the list
+4.  F.A.K.E.J.A.R.V.I.S. deletes the grade of the student's assignment
```
Comment (wesho1107) [inline docs/DeveloperGuide.md]: Any extensions for when grade is not assigned in the first place?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -398,6 +480,88 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case resumes at step 2.
+
+**Use case: Comment on an assignment**
+
+**MSS**
+
+1.  User requests to list persons
+2.  F.A.K.E.J.A.R.V.I.S. shows a list of persons
+3.  User requests to comment on an assignment for a specific person in the list
+4.  F.A.K.E.J.A.R.V.I.S. assigns given comment to the student's assignment
```
Comment (wesho1107) [inline docs/DeveloperGuide.md]: Is there an overwriting of comment if there already exists one?

File: docs/diagrams/CommentClass.puml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+-comment: Comment
++getComment(): Comment
++commentStatus(): boolean
+}
+
+class Comment {
+-commentBody: String
+-isCommented: boolean
++getIsCommented(): boolean
++toString(): String
```
Comment (wesho1107) [inline docs/diagrams/CommentClass.puml:16]: Is toString() important? I think can omit less important details

File: docs/diagrams/GradeClass.puml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+@startuml
+skinparam arrowThickness 1.1
+skinparam classBackgroundColor LOGIC_COLOR
+skinparam classAttributeIconSize 0
```
Comment (wesho1107) [inline docs/diagrams/GradeClass.puml:4]: Perhaps we could standardise on the format, as yours has the circle icon indicating a Class.

**Review by wesho1107 — APPROVED** [review]
Other than a few nits, LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Originalidk) [reply docs/DeveloperGuide.md]: Grade will be pre-assigned as ungraded

File: docs/diagrams/CommentClass.puml (hunk 3 above)
Comment (Originalidk) [reply docs/diagrams/CommentClass.puml:16]: Is the getter method for the field

**Pull request merged**